src/lib.rs

```diff
@@ -505,7 +512,14 @@ enum Repr {
     ArcStr(Arc<str>),
     ArcString(Arc<String>),
     StaticStr(&'static str),
-    Inline { len: u8, buf: [u8; INLINE_CAP] },
+    Inline(Inline),
+}
+
+#[derive(Clone, Copy)]
+struct Inline {
+    _pad: u64,
+    len: u8,
```

Comment:
Try using `len: u64` or `len: u8` with `_padding: [u8; 7]`, or even adding `#[repr(packed)]` without padding?

Reply:
After change the length to usize, this problem seems to be fixed.